### PR TITLE
implement HTMLOutputElement.value

### LINKED ISSUE #25002: Setting HTMLOutputElement's .value doesn't seem to be changing its .innerHTML
This is being picked up by WPT custom-elements/reactions/HTMLOutputElement.html but seems unrelated to the custom element reactions that are the subject of the test.

### PR DESCRIPTION
**Pull request by pshaughn**
HTMLOutputElement now works. I believe layout as well as script is working for it, since it works by putting strings into a child Text node and lays itself out exactly like a span, not requiring any special case rendering like textareas or input elements.

Remaining failures are one mutation test case that everyone else is also failing and probably represents a test/spec mismatch, and all the validation-related tests since we don't have any form-validation hooks.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25002 (GitHub issue number if applicable)

- [X] There are tests for these changes

### CODE REVIEW

**Comment by pshaughn** [comment]
I am certain this is passing more tests than I've already fixed metadata for.
@bors-servo try=wpt

File: components/script/dom/htmloutputelement.rs [hunk 1]
```diff
@@ -68,6 +77,43 @@ impl HTMLOutputElementMethods for HTMLOutputElement {
     fn Labels(&self) -> DomRoot<NodeList> {
         self.upcast::<HTMLElement>().labels()
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-output-defaultvaleu
```
Comment (Manishearth) [inline components/script/dom/htmloutputelement.rs]:
```suggestion
    // https://html.spec.whatwg.org/multipage/#dom-output-defaultvalue
```

File: components/script/dom/node.rs [hunk 2]
```diff
@@ -2017,6 +2017,16 @@ impl Node {
         parent.owner_doc().remove_script_and_layout_blocker();
     }
 
+    // https://dom.spec.whatwg.org/multipage/#string-replace-all
```
Comment (Manishearth) [inline components/script/dom/node.rs]:
```suggestion
    // https://dom.spec.whatwg.org/#string-replace-all
```

only the HTML spec uses `multipage`

File: components/script/dom/node.rs [hunk 3]
```diff
@@ -2017,6 +2017,16 @@ impl Node {
...
+    pub fn string_replace_all(string: DOMString, parent: &Node) {
```
Comment (Manishearth) [inline components/script/dom/node.rs]: Should this be a method on `node`?

**Review by Manishearth — APPROVED** [review]

File: components/script/dom/node.rs (hunk 3 above)
Comment (pshaughn) [reply components/script/dom/node.rs]: I went for consistency with replace_all: https://github.com/servo/servo/blob/728133611656f37480b96103dcc1d025f7d12ba3/components/script/dom/node.rs#L1973-L1974

Comment (Manishearth) [reply components/script/dom/node.rs]: Oh, wait, I thought it wasn't on Node, and then was suggesting that it should be. I didn't notice that it was :)

**Comment by pshaughn** [comment]
Oh, this was just that Mac filter-intermittents python thing...
@bors-servo retry

**Comment by jdm** [comment]
No need to run a full try. We can retry the merge instead.
@bors-servo try- retry

**Comment by jdm** [comment]
@bors-servo retry
* #24726

**Comment by jdm** [comment]
@bors-servo retry
* #25237

**Comment by jdm** [comment]
This can merge with a rebase.

**Comment by pshaughn** [comment]
Rebasing now.

**Comment by jdm** [comment]
@bors-servo retry
* #22212

**Comment by jdm** [comment]
@bors-servo retry
* #22212

**Comment by jdm** [comment]
@bors-servo retry
* #22212

**Pull request merged**